File: packages/android/src/main/java/com/kozmos/components/BrowseCategoriesPanel/BrowseCategoriesPanel.figma.kt
package com.kozmos.components.browsecategoriespanel

import androidx.compose.runtime.Composable
import com.figma.code.connect.Figma
import com.figma.code.connect.FigmaConnect
import com.figma.code.connect.FigmaProperty
import com.figma.code.connect.FigmaType

@FigmaConnect("https://figma.com/design/Yj4O8p6Y9h2Sa9zJVoAiVY?node-id=1351-8026")
class KozmosBrowseCategoriesPanelConnect {
    @FigmaProperty(FigmaType.Text, "Panel Label Text")
    val label: String = "Browse categories"

    @Composable
    fun ComponentExample() {
        KozmosBrowseCategoriesPanel(
            categories = emptyList(),
            onSelect = {},
            label = label
        )
    }
}

File: packages/ios/Sources/Components/BrowseCategoriesPanel/BrowseCategoriesPanel.figma.swift
import SwiftUI
import Figma

struct KozmosBrowseCategoriesPanelConnect: FigmaConnect {
    let component = KozmosBrowseCategoriesPanel<Image, EmptyView, EmptyView, EmptyView>.self
    let figmaNodeUrl = "https://figma.com/design/Yj4O8p6Y9h2Sa9zJVoAiVY?node-id=1351-8026"

    @FigmaString("Panel Label Text")
    var label: String = "Browse categories"

    var body: some View {
        KozmosBrowseCategoriesPanel(
            categories: [],
            label: self.label,
            onSelect: { _ in },
            renderIcon: { _ in Image(systemName: "fork.knife") },
            search: { EmptyView() },
            actions: { EmptyView() },
            emptyState: { EmptyView() }
        )
    }
}
